File: frameworks/go/gorilla-mux/UnitTests/authorized_test.go
package unittests

import (
	"net/http"
	"testing"
)

// rb:test authorized.allowed,authorized.denied
func TestAuthorized(t *testing.T) {
	t.Run("authorized.allowed", func(t *testing.T) {
		response := get(t, "/authorized/small", "Authorization", "Bearer 5a7cc77ed0dcb825806b6f872026c317")

		assertOK(t, payload(t, "items.small.json"), response)
	})

	t.Run("authorized.denied", func(t *testing.T) {
		response := get(t, "/authorized/small", "Authorization", "Bearer 5a7cc77ed0dcb825806b6f872026c310")

		assertStatus(t, response, http.StatusForbidden)
		if body := string(bodyOf(t, response)); body != "Forbidden\n" {
			t.Fatalf("http.Error writes its text and a newline, got %q", body)
		}
	})
}

func TestNoTokenIsRefused(t *testing.T) {
	assertStatus(t, get(t, "/authorized/small"), http.StatusForbidden)
}

File: frameworks/node/express/UnitTests/authorized.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";

import request from "supertest";

import { app, expected } from "./app.ts";

const settings = expected.json("settings.json") as { token: string; wrongToken: string };

// rb:test authorized.allowed
test("authorized.allowed: the accepted token reaches the handler", async () => {
  const response = await request(app).get("/authorized/small").set("authorization", `Bearer ${settings.token}`);

  assert.equal(response.status, 200);
  assert.deepEqual(response.body, expected.json("items.small.json"));
});

// rb:test authorized.denied
test("authorized.denied: a token one character off is refused with res.sendStatus's 403", async () => {
  const response = await request(app).get("/authorized/small").set("authorization", `Bearer ${settings.wrongToken}`);

  assert.equal(response.status, 403);
  assert.match(response.headers["content-type"]!, /^text\/plain/);
  assert.equal(response.text, "Forbidden");
});

test("no token is refused the same way", async () => {
  const response = await request(app).get("/authorized/small");

  assert.equal(response.status, 403);
});
